src/app/api/rest/data-sources/[dataSourceId]/geojson/route.ts

```diff
@@ -0,0 +1,153 @@
+import { findOrganisationForUser } from "@/server/repositories/Organisation";
+import { findDataRecordsByDataSource } from "@/server/repositories/DataRecord";
+import type { DataRecord } from "@/server/models/DataRecord";
+
+/**
+ * Authenticated GeoJSON REST API for data source items
+ * 
+ * GET /api/rest/data-sources/:dataSourceId/geojson
+ * 
+ * Authentication: Basic Auth (user:password)
+ * Returns: GeoJSON FeatureCollection with geocoded items
+ */
+export async function GET(
+  request: NextRequest,
+  { params }: { params: Promise<{ dataSourceId: string }> }
+) {
+  const { dataSourceId } = await params;
+
+  // Check Basic Auth
+  const authHeader = request.headers.get("authorization");
+  if (!authHeader || !authHeader.startsWith("Basic ")) {
+    return new NextResponse(
+      JSON.stringify({ error: "Missing or invalid Authorization header" }),
+      {
+        status: 401,
+        headers: {
+          "WWW-Authenticate": 'Basic realm="Data Source API"',
+          "Content-Type": "application/json",
+        },
+      }
+    );
+  }
+
+  // Decode Basic Auth credentials
+  const base64Credentials = authHeader.split(" ")[1];
+  const credentials = Buffer.from(base64Credentials, "base64").toString("utf-8");
+  const [email, password] = credentials.split(":");
+
+  if (!email || !password) {
+    return new NextResponse(
+      JSON.stringify({ error: "Invalid credentials format" }),
+      {
+        status: 401,
+        headers: {
+          "WWW-Authenticate": 'Basic realm="Data Source API"',
+          "Content-Type": "application/json",
+        },
+      }
+    );
+  }
+
+  // Authenticate user with email and password
+  const user = await findUserByEmailAndPassword({ email, password });
+  if (!user) {
+    return new NextResponse(
+      JSON.stringify({ error: "Invalid credentials" }),
+      {
+        status: 401,
+        headers: {
+          "WWW-Authenticate": 'Basic realm="Data Source API"',
+          "Content-Type": "application/json",
+        },
+      }
+    );
+  }
+
+  // Find data source
+  const dataSource = await findDataSourceById(dataSourceId);
+  if (!dataSource) {
+    return new NextResponse(
+      JSON.stringify({ error: "Data source not found" }),
+      {
+        status: 404,
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+  }
+
+  // Check if user has access to the data source via organization
+  const organisation = await findOrganisationForUser(
+    dataSource.organisationId,
+    user.id
+  );
+  if (!organisation) {
+    return new NextResponse(
+      JSON.stringify({ error: "Access denied" }),
+      {
+        status: 403,
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+  }
+
+  // Fetch all data records for this data source
+  const records = await findDataRecordsByDataSource(
+    dataSourceId,
+    null, // no filter
+    null, // no search
+    0, // page 0
+    [], // no sorting
+    true // get all records
+  );
```

Comment:
@cursoragent can you add query params for filters, search and sorting based on the findDataRecordsByDataSource typing? Use Zod to check it all.